effektif-workflow-api/src/main/java/com/effektif/workflow/api/condition/Not.java

```diff
@@ -40,17 +40,16 @@ public boolean isEmpty() {
 
   @Override
   public void readBpmn(BpmnReader r) {
-    for (XmlElement andElement : r.readElementsEffektif(getClass())) {
-      r.startElement(andElement);
-      condition = r.readCondition();
-      r.endElement();
-    }
+    condition = r.readCondition();
+    r.endElement();
   }
 
   @Override
   public void writeBpmn(BpmnWriter w) {
     w.startElementEffektif(getClass());
-    condition.writeBpmn(w);
+    if (condition != null) {
+      condition.writeBpmn(w);
+    }
```

Comment:
If the condition can be null, isn't it better to not write the element at all?

This looks like an example of something our code doesn't make explicit: only write the element if `this` is ‘valid’. Do you think it would be helpful to add code that either enforces or checks those kinds of invariants?